sdk/core/azure-core/src/main/java/com/azure/core/implementation/serializer/HttpResponseBodyDecoder.java

```diff
@@ -369,9 +369,21 @@ public static boolean shouldEagerlyReadResponse(Type returnType) {
             return false;
         }
 
+
         return isReturnTypeDecodable(returnType)
             || TypeUtil.isTypeOrSubTypeOf(returnType, Void.TYPE)
-            || TypeUtil.isTypeOrSubTypeOf(returnType, Void.class);
+            || TypeUtil.isTypeOrSubTypeOf(returnType, Void.class)
+            || isMonoVoid(returnType);
+    }
+
+    private static boolean isMonoVoid(Type type) {
+        if (TypeUtil.isTypeOrSubTypeOf(type, Mono.class)) {
+            Type monoType = TypeUtil.getTypeArgument(type);
+            return TypeUtil.isTypeOrSubTypeOf(monoType, Void.TYPE)
+                || TypeUtil.isTypeOrSubTypeOf(monoType, Void.class);
+        }
```

Comment:
Does this also need to handle Mono<Response<Void>>?

Reply:
yeah, there are Responses that don't need body.